validator/rules/executor.go

```diff
@@ -95,46 +91,23 @@ func (config *RuleConfig) validate(cmd *cobra.Command, info *validator.StatusLog
 
 // initDefaultRules initialize default rules
 // and overrides the default rules if RuleConfig is provided by the user
-func (config *RuleConfig) initDefaultRules() {
-
-	// default config for rules
-	var defaultConfig = &RuleConfig{
-		Verbose: false,
-		Length: Length{
-			Limits: map[string]Limit{
-				"Use":     {Min: 2},
-				"Short":   {Min: 15},
-				"Long":    {Min: 50},
-				"Example": {Min: 50},
-			},
-		},
-		MustExist: MustExist{
-			Fields: []string{"Use", "Short", "Long", "Example"},
+func (config *RuleConfig) initDefaultRules(cmd *cobra.Command) {
+
+	defaultConfig := RuleConfig{
+		Rules: []Rules{
+			&Length{
+				Verbose: false,
+				Limits: map[string]Limit{
+					"Use":     {Min: 2},
+					"Short":   {Min: 15},
+					"Long":    {Min: 50},
+					"Example": {Min: 50},
+				}},
+			&MustExist{Verbose: false, Fields: []string{"Use", "Short", "Long", "Example"}},
 		},
 	}
 
-	// Check verbose input from user
-	var verbose bool
-	if config != nil && config.Verbose {
-		verbose = true
-	}
-
-	// Set Config to defaultConfig
-	*config = *defaultConfig
-	config.Verbose = verbose
-
-	// Merge the defaultConfig and Config given by user
-	if config.Length.Limits != nil && config.MustExist.Fields != nil {
-		out, err := json.Marshal(config)
-		if err != nil {
-			log.Fatal(err)
-		}
-		data := []byte(out)
-
-		errr := json.Unmarshal(data, &defaultConfig)
-		if errr != nil {
-			log.Fatal(errr)
-		}
-	}
+	// TODO: Override default configuration
```

Comment:
@ankithans All I'm seeing is an empty TODO, have/ can you push up your implementation?

Reply:
sure give me sometime, I will ping you once done. i want to push some working code.

Reply:
It's not essential for it to be working, just change the PR to a [WIP]. I enjoy that you don't want to push non functional code, but if you are still facing issues push it and I'll see if I can help. Don't get too hung up on a single piece of the puzzle. 👍

Reply:
thank you so much!!!
Now Length rule is working fine! as it has a map field Limits to merge
but MustExist cannot be merged with mergo as it is a slice. One solution is convert 
```go
Fields []string
```
to
```go
Fields map[string]bool
```

Reply:
I did it this way, changed the Fields []string to Fields map[string]bool. This way user can disable the default MustExist fields(which was not possible previously)